### PR TITLE
libargon2: init at 20161029

### PR DESCRIPTION
**Pull request by Radvendii**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libargon2/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "021g8wi4g67ywm8zf3yncqwrmfz7ypgm1ih9wcmnxip5n75rymh5";
+  };
+
+  patches = [ ./pkgconfig.patch ];
+
+  installPhase = ''
+    mkdir -p $out
+    make install PREFIX=$out
+    mkdir -p $out/lib/pkgconfig
+    cat <(echo -e "prefix=$out\nversion=${version}") libargon2.pc > $out/lib/pkgconfig/libargon2.pc
```
Comment (0xABAB) [inline pkgs/development/libraries/libargon2/default.nix]: This seems overly complicated. Please rewrite in a simpler fashion.

Comment (Mic92) [inline pkgs/development/libraries/libargon2/default.nix]: how?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -11889,6 +11889,8 @@ with pkgs;
 
   libaio = callPackage ../os-specific/linux/libaio { };
 
+  libargon2 = callPackage ../development/libraries/libargon { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: you named the directory `libargon2`.

File: pkgs/development/libraries/libargon2/pkgconfig.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+-prefix=/usr
+ exec_prefix=${prefix}
+-libdir=${prefix}/lib/@HOST_MULTIARCH@
++libdir=${prefix}/lib/
+ includedir=${prefix}/include
+ 
+ Name: libargon2
+ Description: Development libraries for libargon2
+-Version: @UPSTREAM_VER@
++Version: ${version}
```
Comment (Mic92) [inline pkgs/development/libraries/libargon2/pkgconfig.patch]: You can achieve the same without a patch by using:

```
substitute libargon2.pc  $out/lib/pkgconfig/libargon2.pc \
   --replace @UPSTREAM_VER "${version}"
   --replace @HOST_MULTIARCH@ ""
```

during install.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (Radvendii) [inline pkgs/top-level/all-packages.nix]: Thanks. Fixed.

File: pkgs/development/libraries/libargon2/pkgconfig.patch (hunk 3 above)
Comment (Radvendii) [inline pkgs/development/libraries/libargon2/pkgconfig.patch]: Cool, thanks. I'll fix that. How is the prefix variable supposed to get set though? I don't see other packages having to deal with .pc files at all.

File: pkgs/development/libraries/libargon2/default.nix (hunk 1 above)
Comment (Radvendii) [inline pkgs/development/libraries/libargon2/default.nix]: Working on it. I'm not very familiar with pkgconfig though, and I don't see other packages dealing with this at all, so I'm not sure what to do.

File: pkgs/development/libraries/libargon2/pkgconfig.patch (hunk 3 above)
Comment (Mic92) [inline pkgs/development/libraries/libargon2/pkgconfig.patch]: usually '@variables@' are replaced by the builds-system (autotools) during installation

Comment (Radvendii) [inline pkgs/development/libraries/libargon2/pkgconfig.patch]: That doesn't seem to be done for this one.
Also, presumably I still need to patch the prefix=/usr to say prefix=@PREFIX@ or some such thing, right?

Comment (Mic92) [inline pkgs/development/libraries/libargon2/pkgconfig.patch]: just do:

```
substitute libargon2.pc  $out/lib/pkgconfig/libargon2.pc \
   --replace @UPSTREAM_VER@ "${version}" \
   --replace @HOST_MULTIARCH@ "" \
   --replace 'prefix=/usr' "prefix=/$out"
```

**Comment by Radvendii** [comment]
Alright. I think I've cleaned up the pkgconfig part, but it very well might be possible to make it a lot cleaner still. I don't know.

**Comment by Radvendii** [comment]
Gah! I was trying to rebase and now one of the commits from earlier on showed up here. Help?

**Comment by Mic92** [comment]
you probably rebased against your local master instead of the remote master.

**Pull request merged**